Bugsnag/Payload/BugsnagUser.m

```diff
@@ -43,4 +38,14 @@ - (NSDictionary *)toJson {
     return [NSDictionary dictionaryWithDictionary:dict];
 }
 
+- (BugsnagUser *)withId {
+    if (self.id) {
+        return self;
+    } else {
+        return [[BugsnagUser alloc] initWithId:[BSG_KSSystemInfo deviceAndAppHash]
+                                          name:self.name
+                                  emailAddress:self.email];
+    }
+}
```

Comment:
Why can't we just set this in the constructor when nil is passed as the user ID?

Reply:
I've left it this way to be consistent with the Android notifier where after calling `Bugsnag.setUser(null, null, null)`, `Bugsnag.getUser().id` will be null but events & sessions will have the fallback value.

It's probably not _that_ important a behaviour, but I thought consistency would be best.